
From: <stuart@cybertherial.com>

Hardware Environment: AMD64, sil3512 controller on m/b.  Harddrive: Seagate
120GB ST3120026AS

Problem Description: Drive would lock when writing large files, eg video
from firewire, audio editing.  Checked dmesg, was obviously the sata drive.

Have added the drive to the blacklist in sata_sil.c and the problem seems
to have disappeared.  

Is there a good test to verify that the mod15 bug is really the problem? 

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/scsi/sata_sil.c |    1 +
 1 files changed, 1 insertion(+)

diff -puN drivers/scsi/sata_sil.c~sata_sil-mod15-quirk-with-seagate-st3120026as drivers/scsi/sata_sil.c
--- 25/drivers/scsi/sata_sil.c~sata_sil-mod15-quirk-with-seagate-st3120026as	Thu Oct  7 14:01:18 2004
+++ 25-akpm/drivers/scsi/sata_sil.c	Thu Oct  7 14:01:18 2004
@@ -87,6 +87,7 @@ struct sil_drivelist {
 	{ "ST380023AS",		SIL_QUIRK_MOD15WRITE },
 	{ "ST3120023AS",	SIL_QUIRK_MOD15WRITE },
 	{ "ST3160023AS",	SIL_QUIRK_MOD15WRITE },
+	{ "ST3120026AS",	SIL_QUIRK_MOD15WRITE },
 	{ "ST340014ASL",	SIL_QUIRK_MOD15WRITE },
 	{ "ST360014ASL",	SIL_QUIRK_MOD15WRITE },
 	{ "ST380011ASL",	SIL_QUIRK_MOD15WRITE },
_
